### ISSUE TITLE
Run selection with F8 causes executed code to be recalled into the integrated console prompt

### ISSUE THREAD

**Issue by dbrennand** [PowerShell/vscode-powershell#4041]
### Prerequisites

- [X] I have written a descriptive issue title.
- [X] I have searched all [issues](https://github.com/PowerShell/vscode-powershell/issues?q=is%3Aissue) to ensure it has not already been reported.
- [X] I have read the [troubleshooting](https://github.com/PowerShell/vscode-powershell/blob/master/docs/troubleshooting.md) guide.
- [X] I am sure this issue is with the _extension itself_ and does not reproduce in a standalone [PowerShell](https://github.com/PowerShell/PowerShell/issues/new/choose) instance.
- [X] I have verified that I am using the latest version of Visual Studio Code and the PowerShell extension.
- [X] If this is a security issue, I have read the [security issue reporting guidance](https://github.com/PowerShell/vscode-powershell/blob/master/SECURITY.md).

### Summary

My `vscode-powershell` extension updated this morning to `ms-vscode.powershell@2022.6.1` and I'm having an issue where the selection of code I'm running with F8 is executed but then is recalled immediately into the integrated console prompt.

Example:

```PowerShell
=====> PowerShell Integrated Console v2022.6.1 <=====

PS> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.2.2
PSEdition                      Core
GitCommitId                    7.2.2
OS                             Microsoft Windows 10.0.19042
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

PS> Write-Output -InputObject "Hi!"
Hi!
PS> Write-Output -InputObject "Hi!"
```

It's as if the console history is being immediately recalled into the console prompt without additional input.

Rolling back to `ms-vscode.powershell@2022.5.1` fixes the issue:

```PowerShell
=====> PowerShell Integrated Console v2022.5.1 <=====

PS> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.2.2
PSEdition                      Core
GitCommitId                    7.2.2
OS                             Microsoft Windows 10.0.19042
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

PS> Write-Output -InputObject "Hi!"
Hi!
PS>
```

The latest preview extension `ms-vscode.powershell-preview@2022.6.2` exhibits the same behaviour as  `v2022.6.1`.

Could this be caused by https://github.com/PowerShell/PowerShellEditorServices/pull/1823?

### PowerShell Version

```console
PS> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.2.2
PSEdition                      Core
GitCommitId                    7.2.2
OS                             Microsoft Windows 10.0.19042
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visual Studio Code Version

```console
PS> code --version
1.68.1
30d9c6cd9483b2cc586687151bcbcd635f373630
x64
```

### Extension Version

```console
PS> code --list-extensions --show-versions | Select-String powershell

ms-vscode.powershell@2022.6.1
ms-vscode.powershell-preview@2022.6.2
```

### Steps to Reproduce

```PowerShell
Write-Output -InputObject "Hi!"
```

1. With `ms-vscode.powershell@2022.6.1` installed, launch the PowerShell integrated console.
2. Select line(s) of code in the editor window and run using F8.
3. The code is executed.
4. The executed lines of code are recalled immediately into the integrated console prompt.

**Comment by cblackuk** [comment]
Thanks @dbrennand  - I am having the same issue on multiple systems.

**Comment by andschwa** [comment]
I think you're probably right about it being https://github.com/PowerShell/PowerShellEditorServices/pull/1823. /cc @SeeminglyScience

**Comment by andschwa** [comment]
I'm having trouble reproducing this. Can you share more environment info? @SeeminglyScience sees it only after first causing an error in the PSIC, but neither of you mentioned this. I'm using PS 7.2.4 on macOS.

**Comment by andschwa** [comment]
Ok seems to be Windows-only? Repro'd (without throwing an error first) on PS 7.2.4 on Windows.

**Comment by cblackuk** [comment]
To me it was on a Windows Sever 2019 and WS2022 - brand new session, just after upgrading - the F8 button started pasting the code "twice" as @dbrennand explained.

```powershell
Name                           Value
----                           -----
PSVersion                      7.2.4
PSEdition                      Core
GitCommitId                    7.2.4
OS                             Microsoft Windows 10.0.17763
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

And 

```powershell
Name                           Value
----                           -----
PSVersion                      7.2.5
PSEdition                      Core
GitCommitId                    7.2.5
OS                             Microsoft Windows 10.0.20348
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

```

**Comment by dbrennand** [comment]
Reproduced on a different Windows system than I initially submitted the issue on:

```PowerShell
=====> PowerShell Integrated Console v2022.6.1 <=====

PS C:\Users\dab> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.2.4
PSEdition                      Core
GitCommitId                    7.2.4
OS                             Microsoft Windows 10.0.19043
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

PS C:\Users\dab> 
PS C:\Users\dab> Write-Output -InputObject "Hi!"
Hi!
PS C:\Users\dab> Write-Output -InputObject "Hi!"
```

I can also reproduce this on Linux with WSL on the same system as above:

```PowerShell
=====> PowerShell Integrated Console v2022.6.1 <=====

PS /home/dab> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.2.5
PSEdition                      Core
GitCommitId                    7.2.5
OS                             Linux 5.10.102.1-microsoft-standard-WSL2 #1 SMP Wed Mar 2 00:30:59 UTC 2022
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0

PS /home/dab> 
PS /home/dab> Write-Output -InputObject "Hi!"
Hi!
PS /home/dab> Write-Output -InputObject "Hi!"
```

**Comment by andschwa** [comment]
Fix in PR!

**Comment by cblackuk** [comment]
That is so awesome :) thanks for such a quick turnaround :)

**Comment by dbrennand** [comment]
Wow! Awesome! 🚀 Thanks for the quick turnaround! @andschwa @SeeminglyScience!

**Comment by tropez1971** [comment]
Same issue, even running the latest PowerShell Preview v2022.6.2.

If this is fixed, how do I get my hands on it? Very annoying right now.

**Comment by andschwa** [comment]
Working on getting a preview out right now! I got stuck trying to get a new end-to-end test in yesterday, sorry!

**Comment by cblackuk** [comment]
Thanks @andschwa for your efforts :) really appreciate it!

For @tropez1971  and @pascalkrielen  - for now I would suggest manually downgrading to v2022.5.1 which is what I did for now :-) I hope that helps :-)

**Comment by andschwa** [comment]
Yay, glad to hear it @dbrennand! We're looking to move that preview to stable today, give it a good thorough whack of testing!

**Comment by cblackuk** [comment]
Super happy to confirm that v2022.6.3 is working fine now!!! Thanks a lot everyone :D Great effort!

**Comment by tlgill** [comment]
Bless you - I have been trying to figure this out for several days.  Its not show-stopping but just very annoying.  At first it seemed related to the terminal settings in VS Code but nothing I changed or set back to default helped.  Reinstalled a bunch of things and updated other modules.  Finally tracked it down to the module and found this article/PR.  I updated the module to v20222.6.3 and can verify the issue is gone!!!
